Dockerfile.dev

```diff
@@ -0,0 +1,64 @@
+### Build and install packages
+FROM python:3.6 as build-python
+
+ARG STATIC_URL
+
+ARG SKIP_LOCK
+
+ARG HOST_UID
+ARG HOST_GID
+
+RUN \
+    apt-get -y update && \
+    apt-get install -y gettext && \
+    # Cleanup apt cache
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install Python dependencies
+RUN pip install pipenv
+ADD Pipfile /app/
+ADD Pipfile.lock /app/
+WORKDIR /app
+RUN pipenv install $SKIP_LOCK --system --deploy --dev
+
+
+### Final image
+FROM python:3.6-slim
+
+ARG STATIC_URL
+ENV STATIC_URL ${STATIC_URL:-/static/}
+
+ARG SKIP_LOCK
+
+ARG HOST_UID
+ARG HOST_GID
+ENV APP_USER=${HOST_UID:-saleor}
```

Comment:
The fallback here is `saleor` which is a username and not a uid.
I thought a uid was a number like `1001` ?
Not 100% myself but please clarify.

Reply:
I'd like to suggest the following slight change:

-  remove APP_USER as is not needed I think but HOST_UID is needed. I am open to correction on this.
- explicit command line arguments for `useradd`, it's easier to read when going through the `Dockerfile` to figure out what is going on
- Provide fallbacks for HOST_UID and HOST_GID, user can provide his/her own though. Don't depend on a makefile exclusively. Not everyone uses makefiles.

```bash
ARG HOST_UID
ARG HOST_GID
ENV HOST_UID=${HOST_UID:-1001} \
    HOST_GID=${HOST_GID:-1001}

RUN useradd --non-unique --uid ${HOST_UID}  --gid  ${HOST_GID} --create-home saleor && \
    install -d -o ${HOST_UID} -g ${HOST_GID} /app/node_modules && \
    install -d -o ${HOST_UID} -g ${HOST_GID} /app/media && \
    install -d -o ${HOST_UID} -g ${HOST_GID} /app/static
```

Reply:
Yes APP_USER is redundant so that make more sense. A fallback uid and gid of 1001 can be used but they will have be set explicitly to the host uid and gid for the configuration to work as gosu requires them. That can be done through the Makefile, a shell script, common.env, docker-compose.dev.yml or on the command line with --build-args=